src/internal/mod.rs

```diff
@@ -729,6 +736,25 @@ fn augment_private_key_with_retry<R: KeyGenOps>(
     aug_private_key().or_else(|_| aug_private_key())
 }
 
+// Subtracts a generated private key from the provided PrivateKey, returning
+// the result and the plaintext associated with the generated key.
+// There is a very small chance that the generated private key could not be compatible with
+// the given PrivateKey, so we retry once internally before giving the caller an error.
+fn gen_plaintext_and_aug_with_retry<R: recrypt::api::CryptoOps + KeyGenOps>(
```

Comment:
Is there an opportunity to combine this with the other augmentation function? Maybe tying augmentation to PrivateKey was the wrong thing? Wondering if you have thoughts here. If we're going to address, we probably should do so in another PR.

Reply:
They likely could in the future, but since one requires `KeyGenOps` to make an `AugmentationFactor` and the other requires `CryptoOps` to generate a plaintext/derive its private key, I'd prefer to leave them separate in this PR.